tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/NUnitIntegration.cs

```diff
@@ -15,366 +15,365 @@
+        {
+            return null;
+        }
 
-            // Get traits
-            Dictionary<string, List<string>>? traits = null;
-            if (testMethodProperties != null)
-            {
-                skipReason = (string)testMethodProperties.Get(SkipReasonKey);
-                ExtractTraits(currentTest, ref traits);
-            }
+        var test = suite.InternalCreateTest(testMethod.Name);
+        string? skipReason = null;
+
+        // Get test parameters
+        var methodParameters = testMethod.GetParameters();
+        if (methodParameters?.Length > 0)
+        {
+            var testParameters = new TestParameters();
+            testParameters.Metadata = new Dictionary<string, object?>();
+            testParameters.Arguments = new Dictionary<string, object>();
+            testParameters.Metadata[TestTags.MetadataTestName] = currentTest.Name ?? string.Empty;
 
-            if (traits?.Count > 0)
+            for (int i = 0; i < methodParameters.Length; i++)
             {
-                // Unskippable test
-                if (CIVisibility.Settings.IntelligentTestRunnerEnabled)
+                var key = methodParameters[i].Name ?? string.Empty;
+                if (testMethodArguments != null && i < testMethodArguments.Length)
                 {
-                    ShouldSkip(currentTest, out var isUnskippable, out var isForcedRun, traits);
-                    test.SetTag(IntelligentTestRunnerTags.UnskippableTag, isUnskippable ? "true" : "false");
-                    test.SetTag(IntelligentTestRunnerTags.ForcedRunTag, isForcedRun ? "true" : "false");
-                    traits.Remove(IntelligentTestRunnerTags.UnskippableTraitName);
+                    testParameters.Arguments[key] = Common.GetParametersValueData(testMethodArguments[i]);
                 }
-
-                test.SetTraits(traits);
-            }
-            else
-            {
-                // Unskippable test
-                if (CIVisibility.Settings.IntelligentTestRunnerEnabled)
+                else
                 {
-                    test.SetTag(IntelligentTestRunnerTags.UnskippableTag, "false");
-                    test.SetTag(IntelligentTestRunnerTags.ForcedRunTag, "false");
+                    testParameters.Arguments[key] = "(default)";
                 }
             }
 
-            // Early flake detection flags
-            Common.SetEarlyFlakeDetectionTestTagsAndAbortReason(test, isRetry, ref _newTestCases, ref _totalTestCases);
-
-            // Flaky Retry
-            Common.SetFlakyRetryTags(test, isRetry);
+            test.SetParameters(testParameters);
+        }
 
-            // Test code and code owners
-            test.SetTestMethodInfo(testMethod);
+        // Get traits
+        Dictionary<string, List<string>>? traits = null;
+        if (testMethodProperties != null)
+        {
+            skipReason = (string?)testMethodProperties.Get(SkipReasonKey);
```

Comment:
super nit- is there any reason not to use `as` now, seeing as it's safer?
```suggestion
            skipReason = testMethodProperties.Get(SkipReasonKey) as string;
```

Reply:
true, done in [3d98a45](https://github.com/DataDog/dd-trace-dotnet/pull/6374/commits/3d98a45a5bebdc0c7de9990b062d3e8eb506713d)